azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/servicebus/ServiceBusAutoConfiguration.java

```diff
@@ -66,13 +69,24 @@ public SubscriptionClient subscriptionClient() throws ServiceBusException, Inter
                 properties.getSubscriptionReceiveMode());
     }
 
+    private String getHashNamespace() {
+        final String namespace = properties.getConnectionString()
+                .replaceFirst("^.*//", "") // emit head 'Endpoint=sb://'
+                .replaceAll("\\..*$", ""); // emit tail '${namespace}.xxx.xxx'
+
+        // Namespace can only be letter, number and hyphen, start with letter, end with letter or number,
+        // with length of 6-50.
```

Comment:
It is possible that the naming constraint will be extended? Should we stop user from here?

Reply:
make sense, better to just warning here.